pkg/apis/core/validation/validation.go

```diff
@@ -3008,12 +3008,19 @@ func ValidateNodeSelectorRequirement(rq core.NodeSelectorRequirement, fldPath *f
 func ValidateNodeSelectorTerm(term core.NodeSelectorTerm, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	if len(term.MatchExpressions) == 0 {
-		return append(allErrs, field.Required(fldPath.Child("matchExpressions"), "must have at least one node selector requirement"))
+	if len(term.MatchExpressions) == 0 && len(term.MatchFields) == 0 {
```

Comment:
I would actually remove this requirement completely... an empty term matches no objects. Stick to validating MatchExpressions or MatchFields if they provide them

Reply:
sure, will handle it in https://github.com/kubernetes/kubernetes/pull/62448